src/expr.rs

```diff
@@ -455,9 +454,19 @@ fn rewrite_closure(capture: ast::CaptureBy,
 
     // We couldn't format the closure body as a single line expression; fall
     // back to block formatting.
-    let body_rewrite = inner_block.rewrite(&context, budget, Indent::empty());
+    let body_rewrite = try_opt!(inner_block.rewrite(&context, budget, Indent::empty()));
 
-    Some(format!("{} {}", prefix, try_opt!(body_rewrite)))
+    let block_threshold = context.config.closure_block_indent_threshold;
```

Comment:
This cut-off seems fairly arbitrary. Can we consider a "try-both-and-take-the-more-compact-one" heuristic?